okio/src/commonMain/kotlin/okio/internal/Buffer.kt

```diff
@@ -1408,6 +1408,22 @@ internal inline fun Buffer.commonHashCode(): Int {
   return result
 }
 
+internal inline fun Buffer.commonCopy(): Buffer {
+  val result = Buffer()
+  if (size == 0L) return result
+
+  result.head = head!!.sharedCopy()
```

Comment:
This problem persists in a few places, but I'd love to see locals being leveraged here to avoid so much bang-bangs.
```kotlin
val myHead = head!!
val copyHead = myHead.sharedCopy()
// .. use myHead and copyHead without !!
```
Ideally we bang-bang each unique reference once. It clutters up the source code a bit, but has _zero_ impact on bytecode.

(I mean it actually has a negative size impact on bytecode since we don't emit redundant calls to `Intrinsics.checkNotNull` et. al., but compared to Java is has zero impact)